log/levels/levels.go

```diff
@@ -29,8 +29,54 @@ type Levels struct {
 	warnValue  string
 	errorValue string
 	critValue  string
+
+	// FilterLogLevel is an option allowing logs below a certain level to be
+	// discarded rather than logged. For example, a consumer may set LogLevel
+	// to levels.Warn, and any logs to Debug or Info would become no-ops.
+	filterLogLevel LogLevel
+
+	// Promote errors
+	promoteErrors       bool
+	errorKey            string
+	promoteErrorToLevel LogLevel
+}
+
+// levelCommittedLogger embeds a log level that the user has selected and prepends
+// the associated level string as a prefix to every log line. Log level may be
+// affected by error promotion; and log lines may be suppressed based on the value
+// of FilterLogLevel.
+type levelCommittedLogger struct {
```

Comment:
I would much rather see this as a distinct implementation of the Logger (or Levels) interface, than as a change to the behavior of Levels. Otherwise, systems that currently depend on kitlog behavior will be encouraged (or forced) to fork go-kit.